### PR TITLE
copying from aind-data-schema

### PR DESCRIPTION
**Pull request by dyf**
This copies over existing upgrading methods from aind-data-schema unchanged, except for separating DataDescription and Processing tests into separate files.

Required for https://github.com/AllenNeuralDynamics/aind-data-schema/issues/709

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -4,3 +4,4 @@ exclude =
     __pycache__,
     build
 max-complexity = 10
+max-line-length = 120
```
Comment (jtyoung84) [inline .flake8:7]: Is this what we've settled on? Almost all the other repositories are using the pep standard `79`,

File: pyproject.toml [hunk 2]
```diff
@@ -37,7 +38,7 @@ where = ["src"]
 version = {attr = "aind_metadata_upgrader.__version__"}
 
 [tool.black]
-line-length = 79
+line-length = 120
 target_version = ['py36']
```
Comment (jtyoung84) [inline pyproject.toml:42]: We may need to update the template also. This needs to be py38 and the requires python line above needs to be >=3.8

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
One requested change to update the pyproject.toml file to reflect python >= 3.8 and one suggested change to keep the line length at 79. If you really want 120, then I'll approve it. I went ahead and added the service account user as an admin since it is still required for certain things with github actions. I'll update some of the other settings also.

File: .flake8 (hunk 1 above)
Comment (dyf) [reply .flake8:7]: let's leave this as-is and fix in a subsequent PR, lots of changes needed.

**Comment by dyf** [comment]
found some other upgrade logic to bring over

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**